packages/ts/react-grid/test/autogrid.spec.tsx

```diff
@@ -120,6 +120,13 @@ describe('@hilla/react-grid', () => {
       expect(grid.getAttribute('model')).to.be.null;
       expect(grid.getAttribute('service')).to.be.null;
     });
+    it('calls data provider list() only once for initial data', async () => {
```

Comment:
This test also passes when removing the changes. The only scenario where it triggers less requests is when toggling header filters, though in that case you still get more than one as the sort is applied with a delay again. 

The change probably makes sense overall, but was there some specific case that you were testing with?

Reply:
It seems like the data provider is no longer being called ever with an undefined filter. Let's just add the test so it can catch potential future problems. The actual problem must have been solved after the issue was created. I just rebased an old fix for it